src/python_inspector/utils_pypi.py

```diff
@@ -675,7 +676,7 @@ def from_link(cls, link: Link):
         """
         requires_python = link.python_requires
         path_or_url = link.url
-        filename = os.path.basename(path_or_url.strip("/"))
+        filename = os.path.basename(unquote(path_or_url).strip("/"))
```

Comment:
Are you sure you want to unquote there and not after getting the basename, so this happens just on the filename and not the whole path?

What if cls.`from_filename(filename)` directly with an encoded filename? Could it be a better place then?

Reply:
Makes sense to me. I've changed it, hope it's in the right place (overload) of that function.